modules/apps/captcha/captcha-taglib/src/main/resources/META-INF/resources/captcha/simplecaptcha.jsp

```diff
@@ -41,23 +41,25 @@ String url = (String)request.getAttribute("liferay-captcha:captcha:url");
 	</div>
 
 	<aui:script>
-		var refreshCaptcha = document.getElementById(
-			'<portlet:namespace />refreshCaptcha'
-		);
-
-		if (refreshCaptcha) {
-			refreshCaptcha.addEventListener('click', function() {
-				var url = Liferay.Util.addParams(
-					't=' + Date.now(),
-					'<%= HtmlUtil.escapeJS(url) %>'
-				);
-
-				var captcha = document.getElementById('<portlet:namespace />captcha');
-
-				if (captcha) {
-					captcha.setAttribute('src', url);
-				}
-			});
+		window.onload = function() {
+			var refreshCaptcha = document.getElementById(
+				'<portlet:namespace />refreshCaptcha'
+			);
+
+			if (refreshCaptcha) {
+				refreshCaptcha.addEventListener('click', function() {
+					var url = Liferay.Util.addParams(
+						't=' + Date.now(),
+						'<%= HtmlUtil.escapeJS(url) %>'
+					);
+
+					var captcha = document.getElementById('<portlet:namespace />captcha');
+
+					if (captcha) {
+						captcha.setAttribute('src', url);
+					}
+				});
+			}
```

Comment:
@brunobasto In order for Captcha to get the refresh button, I needed to wrap this inside the `window.onload`, this code was being executed before React started rendering the Captcha component. Another option would be to implement this within the component, but I don't know if we could get the information from the Captcha url.

Reply:
Hey @matuzalemsteles, I think overriding the `window.onload` is bad. It could have unwanted side effects on other people's stuff. Can we using `window.addEventListener('load')`?

Reply:
Oh yeah, good catch.

Reply:
Also, have you tested with other kinds of captcha? Let me see how to configure it